### PR TITLE
Redirect status

### PR DESCRIPTION
**Pull request by kviktor**
Works great with PNG status (for example: https://git.ik.bme.hu/circle/cloud)

### CODE REVIEW

File: master/buildbot/status/web/redirectstatus.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                number = build.getNumber()
+                url = "/builders/%(builder)s/builds/%(number)d" % {
+                    'builder': builder,
+                    'number': number,
+                }
+                return url
+        return "/waterfall"
+
+    def render(self, request):
+        url = self.get_url(request)
```
Comment (sa2ajj) [inline master/buildbot/status/web/redirectstatus.py]: [nit] is this really necessary? why not put `self.get_url(request)` directly to the call below?

File: master/buildbot/status/web/redirectstatus.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        """
+        revision = request.args.get("revision", [None])[0]
+
+        builder = request.args.get("builder", [None])[0]
+
+        if revision and builder in self.status.getBuilderNames():
+            build = self.status.getBuilder(builder).getBuild(0, revision)
+            if build:
+                number = build.getNumber()
+                url = "/builders/%(builder)s/builds/%(number)d" % {
```
Comment (sa2ajj) [inline master/buildbot/status/web/redirectstatus.py]: i'd suggest using `buildbot.status.web.path_to_build` here

File: master/buildbot/status/web/redirectstatus.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                    'builder': builder,
+                    'number': number,
+                }
+                return url
+        return "/waterfall"
```
Comment (sa2ajj) [inline master/buildbot/status/web/redirectstatus.py]: i do not think this is correct

**Comment by sa2ajj** [comment]
Looks good otherwise

File: master/buildbot/status/web/baseweb.py [hunk 4]
```diff
@@ -102,7 +103,9 @@ class WebStatus(service.MultiService):
      /builders/BUILDERNAME/builds/NUM/steps/STEPNAME/logs/LOGNAME: a StatusLog
      /builders/_all/{force,stop}: force a build/stop building on all builders.
      /buildstatus?builder=...&number=...: an embedded iframe for the console
-     /png?buildername=...&number=...&size=... a png resource with build info
+     /png?builder=...&number=...&size=... a png resource with build info
+     /redirect?builder=...&revision=... a redirect to the build that built the
+                                        given revision
```
Comment (jaredgrubb) [inline master/buildbot/status/web/baseweb.py]: "... to the *last* build that built ..." -- technically there could be more than one. (I assume it's last, maybe it's first?)

Comment (kviktor) [inline master/buildbot/status/web/baseweb.py]: True, in case of force build it redirects to the forced one.

**Comment by sa2ajj** [comment]
@kviktor a few changes and we take it in! :)

**Comment by sa2ajj** [comment]
@kviktor any updates?

And a rebase is required.

**Comment by djmitche** [comment]
Is this forward-compatible to nine?  If not, I'm inclined to close, to avoid adding features in eight that aren't replicated in nine.

**Comment by djmitche** [comment]
In fact, given that and over a month with no response, per http://trac.buildbot.net/wiki/StalePullRequestPolicy I'm going to close the PR.  The code is still here!

**Comment by kviktor** [comment]
I don't know about forward-compatibility but function wise it's very similar to pngstatus and can probably be implemented the same way (last time I checked it's not in nine yet tho)

**Pull request closed without merging**